internal/install/_static/terraform_deployer_run.sh

```diff
@@ -2,25 +2,41 @@
 
 set -euo pipefail
 
-# Terraform code may rely on content from other files than .tf files (es json, zip, html, text), so we copy all the content over
-# See more: https://github.com/elastic/elastic-package/pull/603
-cp -r /stage/* /workspace
-
 cleanup() {
   r=$?
 
-  set -x
   terraform destroy -auto-approve
 
   exit $r
 }
-trap cleanup EXIT INT TERM
 
-terraform init
-terraform plan
-terraform apply -auto-approve && touch /tmp/tf-applied
+gcp_auth() {
+  if test -n "$(printenv "GOOGLE_CREDENTIALS")"; then
+    # Save GCP credentials on disk and perform authentication
+    # NOTE: this is required for bq (and maybe other gcloud related tools) to authenticate
+    export "GOOGLE_APPLICATION_CREDENTIALS=/root/.config/gcloud/application_default_credentials.json"
+    printenv "GOOGLE_CREDENTIALS" > "$GOOGLE_APPLICATION_CREDENTIALS"
+    gcloud auth login --cred-file "$GOOGLE_APPLICATION_CREDENTIALS"
+    # NOTE: Terraform support authentication through GOOGLE_CREDENTIALS and usual gcloud ADC but other
+    # tools (like bq) don't support the first, so we always rely on gcloud ADC.
+    unset "GOOGLE_CREDENTIALS"
+  fi
+}
+
+if [[ "${BASH_SOURCE[0]}" = "$0" ]]; then
```

Comment:
Why is this condition required here?

Reply:
It is not required, but it avoid running the code in case the file is `source`d and identify a "main" like are, like the `main` function in go or C. From that point onward all code will be executed only when the file is explicitly executed (`bash file.sh` or `./file.sh`). It helps grouping the relevant "main" code in the same area, preventing arbitrary code being added between functions.

Consider it my personal way of writing BASH files, I think it aids reading them (like `__file__ == __main__` does for Python)

Reply:
I understand your point of view, but please keep it simple as it was before.

Reply:
Is run by container engine but is still read by developers :) The point was to simplifying reading the code flow. Anyway I removed it (will push once rebased onto main)